// @stacks/blockchain-api-client/lib/generated/apis/NonFungibleTokensApi.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json)
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.NonFungibleTokensApi = void 0;
const runtime = require("../runtime");
const models_1 = require("../models");
/**
 *
 */
class NonFungibleTokensApi extends runtime.BaseAPI {
    /**
     * Retrieves all events relevant to a Non-Fungible Token. Useful to determine the ownership history of a particular asset.  More information on Non-Fungible Tokens on the Stacks blockchain can be found [here](https://docs.stacks.co/write-smart-contracts/tokens#non-fungible-tokens-nfts).
     * Non-Fungible Token history
     */
    async getNftHistoryRaw(requestParameters, initOverrides) {
        if (requestParameters.assetIdentifier === null || requestParameters.assetIdentifier === undefined) {
            throw new runtime.RequiredError('assetIdentifier', 'Required parameter requestParameters.assetIdentifier was null or undefined when calling getNftHistory.');
        }
        if (requestParameters.value === null || requestParameters.value === undefined) {
            throw new runtime.RequiredError('value', 'Required parameter requestParameters.value was null or undefined when calling getNftHistory.');
        }
        const queryParameters = {};
        if (requestParameters.assetIdentifier !== undefined) {
            queryParameters['asset_identifier'] = requestParameters.assetIdentifier;
        }
        if (requestParameters.value !== undefined) {
            queryParameters['value'] = requestParameters.value;
        }
        if (requestParameters.limit !== undefined) {
            queryParameters['limit'] = requestParameters.limit;
        }
        if (requestParameters.offset !== undefined) {
            queryParameters['offset'] = requestParameters.offset;
        }
        if (requestParameters.unanchored !== undefined) {
            queryParameters['unanchored'] = requestParameters.unanchored;
        }
        if (requestParameters.txMetadata !== undefined) {
            queryParameters['tx_metadata'] = requestParameters.txMetadata;
        }
        const headerParameters = {};
        const response = await this.request({
            path: `/extended/v1/tokens/nft/history`,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        }, initOverrides);
        return new runtime.JSONApiResponse(response, (jsonValue) => (0, models_1.NonFungibleTokenHistoryEventListFromJSON)(jsonValue));
    }
    /**
     * Retrieves all events relevant to a Non-Fungible Token. Useful to determine the ownership history of a particular asset.  More information on Non-Fungible Tokens on the Stacks blockchain can be found [here](https://docs.stacks.co/write-smart-contracts/tokens#non-fungible-tokens-nfts).
     * Non-Fungible Token history
     */
    async getNftHistory(requestParameters, initOverrides) {
        const response = await this.getNftHistoryRaw(requestParameters, initOverrides);
        return await response.value();
    }
    /**
     * Retrieves the list of Non-Fungible Tokens owned by the given principal (STX address or Smart Contract ID). Results can be filtered by one or more asset identifiers and can include metadata about the transaction that made the principal own each token.  More information on Non-Fungible Tokens on the Stacks blockchain can be found [here](https://docs.stacks.co/write-smart-contracts/tokens#non-fungible-tokens-nfts).
     * Non-Fungible Token holdings
     */
    async getNftHoldingsRaw(requestParameters, initOverrides) {
        if (requestParameters.principal === null || requestParameters.principal === undefined) {
            throw new runtime.RequiredError('principal', 'Required parameter requestParameters.principal was null or undefined when calling getNftHoldings.');
        }
        const queryParameters = {};
        if (requestParameters.principal !== undefined) {
            queryParameters['principal'] = requestParameters.principal;
        }
        if (requestParameters.assetIdentifiers) {
            queryParameters['asset_identifiers'] = requestParameters.assetIdentifiers;
        }
        if (requestParameters.limit !== undefined) {
            queryParameters['limit'] = requestParameters.limit;
        }
        if (requestParameters.offset !== undefined) {
            queryParameters['offset'] = requestParameters.offset;
        }
        if (requestParameters.unanchored !== undefined) {
            queryParameters['unanchored'] = requestParameters.unanchored;
        }
        if (requestParameters.txMetadata !== undefined) {
            queryParameters['tx_metadata'] = requestParameters.txMetadata;
        }
        const headerParameters = {};
        const response = await this.request({
            path: `/extended/v1/tokens/nft/holdings`,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        }, initOverrides);
        return new runtime.JSONApiResponse(response, (jsonValue) => (0, models_1.NonFungibleTokenHoldingsListFromJSON)(jsonValue));
    }
    /**
     * Retrieves the list of Non-Fungible Tokens owned by the given principal (STX address or Smart Contract ID). Results can be filtered by one or more asset identifiers and can include metadata about the transaction that made the principal own each token.  More information on Non-Fungible Tokens on the Stacks blockchain can be found [here](https://docs.stacks.co/write-smart-contracts/tokens#non-fungible-tokens-nfts).
     * Non-Fungible Token holdings
     */
    async getNftHoldings(requestParameters, initOverrides) {
        const response = await this.getNftHoldingsRaw(requestParameters, initOverrides);
        return await response.value();
    }
    /**
     * Retrieves all mint events for a Non-Fungible Token asset class. Useful to determine which NFTs of a particular collection have been claimed.  More information on Non-Fungible Tokens on the Stacks blockchain can be found [here](https://docs.stacks.co/write-smart-contracts/tokens#non-fungible-tokens-nfts).
     * Non-Fungible Token mints
     */
    async getNftMintsRaw(requestParameters, initOverrides) {
        if (requestParameters.assetIdentifier === null || requestParameters.assetIdentifier === undefined) {
            throw new runtime.RequiredError('assetIdentifier', 'Required parameter requestParameters.assetIdentifier was null or undefined when calling getNftMints.');
        }
        const queryParameters = {};
        if (requestParameters.assetIdentifier !== undefined) {
            queryParameters['asset_identifier'] = requestParameters.assetIdentifier;
        }
        if (requestParameters.limit !== undefined) {
            queryParameters['limit'] = requestParameters.limit;
        }
        if (requestParameters.offset !== undefined) {
            queryParameters['offset'] = requestParameters.offset;
        }
        if (requestParameters.unanchored !== undefined) {
            queryParameters['unanchored'] = requestParameters.unanchored;
        }
        if (requestParameters.txMetadata !== undefined) {
            queryParameters['tx_metadata'] = requestParameters.txMetadata;
        }
        const headerParameters = {};
        const response = await this.request({
            path: `/extended/v1/tokens/nft/mints`,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        }, initOverrides);
        return new runtime.JSONApiResponse(response, (jsonValue) => (0, models_1.NonFungibleTokenMintListFromJSON)(jsonValue));
    }
    /**
     * Retrieves all mint events for a Non-Fungible Token asset class. Useful to determine which NFTs of a particular collection have been claimed.  More information on Non-Fungible Tokens on the Stacks blockchain can be found [here](https://docs.stacks.co/write-smart-contracts/tokens#non-fungible-tokens-nfts).
     * Non-Fungible Token mints
     */
    async getNftMints(requestParameters, initOverrides) {
        const response = await this.getNftMintsRaw(requestParameters, initOverrides);
        return await response.value();
    }
}
exports.NonFungibleTokensApi = NonFungibleTokensApi;
