// @stacks/blockchain-api-client/lib/generated/apis/StackingApi.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json)
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.StackingApi = void 0;
const runtime = require("../runtime");
const models_1 = require("../models");
/**
 *
 */
class StackingApi extends runtime.BaseAPI {
    /**
     * Retrieves the list of stacking pool members for a given delegator principal.
     * Stacking pool members
     */
    async getPoolDelegationsRaw(requestParameters, initOverrides) {
        if (requestParameters.poolPrincipal === null || requestParameters.poolPrincipal === undefined) {
            throw new runtime.RequiredError('poolPrincipal', 'Required parameter requestParameters.poolPrincipal was null or undefined when calling getPoolDelegations.');
        }
        const queryParameters = {};
        if (requestParameters.afterBlock !== undefined) {
            queryParameters['after_block'] = requestParameters.afterBlock;
        }
        if (requestParameters.unanchored !== undefined) {
            queryParameters['unanchored'] = requestParameters.unanchored;
        }
        if (requestParameters.limit !== undefined) {
            queryParameters['limit'] = requestParameters.limit;
        }
        if (requestParameters.offset !== undefined) {
            queryParameters['offset'] = requestParameters.offset;
        }
        const headerParameters = {};
        const response = await this.request({
            path: `/extended/beta/stacking/{pool_principal}/delegations`.replace(`{${"pool_principal"}}`, encodeURIComponent(String(requestParameters.poolPrincipal))),
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        }, initOverrides);
        return new runtime.JSONApiResponse(response, (jsonValue) => (0, models_1.PoolDelegationsResponseFromJSON)(jsonValue));
    }
    /**
     * Retrieves the list of stacking pool members for a given delegator principal.
     * Stacking pool members
     */
    async getPoolDelegations(requestParameters, initOverrides) {
        const response = await this.getPoolDelegationsRaw(requestParameters, initOverrides);
        return await response.value();
    }
}
exports.StackingApi = StackingApi;
