// @stacks/blockchain-api-client/src/generated/models/AddressTransactionsV2ListResponseEvents.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    AddressTransactionsV2ListResponseEventsStx,
    AddressTransactionsV2ListResponseEventsStxFromJSON,
    AddressTransactionsV2ListResponseEventsStxFromJSONTyped,
    AddressTransactionsV2ListResponseEventsStxToJSON,
} from './';

/**
 * 
 * @export
 * @interface AddressTransactionsV2ListResponseEvents
 */
export interface AddressTransactionsV2ListResponseEvents {
    /**
     * 
     * @type {AddressTransactionsV2ListResponseEventsStx}
     * @memberof AddressTransactionsV2ListResponseEvents
     */
    stx: AddressTransactionsV2ListResponseEventsStx;
    /**
     * 
     * @type {AddressTransactionsV2ListResponseEventsStx}
     * @memberof AddressTransactionsV2ListResponseEvents
     */
    ft: AddressTransactionsV2ListResponseEventsStx;
    /**
     * 
     * @type {AddressTransactionsV2ListResponseEventsStx}
     * @memberof AddressTransactionsV2ListResponseEvents
     */
    nft: AddressTransactionsV2ListResponseEventsStx;
}

export function AddressTransactionsV2ListResponseEventsFromJSON(json: any): AddressTransactionsV2ListResponseEvents {
    return AddressTransactionsV2ListResponseEventsFromJSONTyped(json, false);
}

export function AddressTransactionsV2ListResponseEventsFromJSONTyped(json: any, ignoreDiscriminator: boolean): AddressTransactionsV2ListResponseEvents {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'stx': AddressTransactionsV2ListResponseEventsStxFromJSON(json['stx']),
        'ft': AddressTransactionsV2ListResponseEventsStxFromJSON(json['ft']),
        'nft': AddressTransactionsV2ListResponseEventsStxFromJSON(json['nft']),
    };
}

export function AddressTransactionsV2ListResponseEventsToJSON(value?: AddressTransactionsV2ListResponseEvents | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'stx': AddressTransactionsV2ListResponseEventsStxToJSON(value.stx),
        'ft': AddressTransactionsV2ListResponseEventsStxToJSON(value.ft),
        'nft': AddressTransactionsV2ListResponseEventsStxToJSON(value.nft),
    };
}
