// @stacks/blockchain-api-client/src/generated/models/AddressUnlockSchedule.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Unlock schedule amount and block height
 * @export
 * @interface AddressUnlockSchedule
 */
export interface AddressUnlockSchedule {
    /**
     * Micro-STX amount locked at this block height.
     * @type {string}
     * @memberof AddressUnlockSchedule
     */
    amount: string;
    /**
     * 
     * @type {number}
     * @memberof AddressUnlockSchedule
     */
    block_height: number;
}

export function AddressUnlockScheduleFromJSON(json: any): AddressUnlockSchedule {
    return AddressUnlockScheduleFromJSONTyped(json, false);
}

export function AddressUnlockScheduleFromJSONTyped(json: any, ignoreDiscriminator: boolean): AddressUnlockSchedule {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'amount': json['amount'],
        'block_height': json['block_height'],
    };
}

export function AddressUnlockScheduleToJSON(value?: AddressUnlockSchedule | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'amount': value.amount,
        'block_height': value.block_height,
    };
}
