// @stacks/blockchain-api-client/src/generated/models/AddressTransactionsV2ListResponseEventsStx.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * 
 * @export
 * @interface AddressTransactionsV2ListResponseEventsStx
 */
export interface AddressTransactionsV2ListResponseEventsStx {
    /**
     * 
     * @type {number}
     * @memberof AddressTransactionsV2ListResponseEventsStx
     */
    transfer: number;
    /**
     * 
     * @type {number}
     * @memberof AddressTransactionsV2ListResponseEventsStx
     */
    mint: number;
    /**
     * 
     * @type {number}
     * @memberof AddressTransactionsV2ListResponseEventsStx
     */
    burn: number;
}

export function AddressTransactionsV2ListResponseEventsStxFromJSON(json: any): AddressTransactionsV2ListResponseEventsStx {
    return AddressTransactionsV2ListResponseEventsStxFromJSONTyped(json, false);
}

export function AddressTransactionsV2ListResponseEventsStxFromJSONTyped(json: any, ignoreDiscriminator: boolean): AddressTransactionsV2ListResponseEventsStx {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'transfer': json['transfer'],
        'mint': json['mint'],
        'burn': json['burn'],
    };
}

export function AddressTransactionsV2ListResponseEventsStxToJSON(value?: AddressTransactionsV2ListResponseEventsStx | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'transfer': value.transfer,
        'mint': value.mint,
        'burn': value.burn,
    };
}
