// @stacks/blockchain-api-client/src/generated/models/AddressTransactionsV2ListResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    AddressTransaction,
    AddressTransactionFromJSON,
    AddressTransactionFromJSONTyped,
    AddressTransactionToJSON,
} from './';

/**
 * GET Address Transactions
 * @export
 * @interface AddressTransactionsV2ListResponse
 */
export interface AddressTransactionsV2ListResponse {
    /**
     * 
     * @type {number}
     * @memberof AddressTransactionsV2ListResponse
     */
    limit: number;
    /**
     * 
     * @type {number}
     * @memberof AddressTransactionsV2ListResponse
     */
    offset: number;
    /**
     * 
     * @type {number}
     * @memberof AddressTransactionsV2ListResponse
     */
    total: number;
    /**
     * 
     * @type {Array<AddressTransaction>}
     * @memberof AddressTransactionsV2ListResponse
     */
    results: Array<AddressTransaction>;
}

export function AddressTransactionsV2ListResponseFromJSON(json: any): AddressTransactionsV2ListResponse {
    return AddressTransactionsV2ListResponseFromJSONTyped(json, false);
}

export function AddressTransactionsV2ListResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): AddressTransactionsV2ListResponse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'limit': json['limit'],
        'offset': json['offset'],
        'total': json['total'],
        'results': ((json['results'] as Array<any>).map(AddressTransactionFromJSON)),
    };
}

export function AddressTransactionsV2ListResponseToJSON(value?: AddressTransactionsV2ListResponse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'limit': value.limit,
        'offset': value.offset,
        'total': value.total,
        'results': ((value.results as Array<any>).map(AddressTransactionToJSON)),
    };
}
