// @stacks/blockchain-api-client/src/generated/models/AddressTransactionEventListResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * GET Address Transaction Events
 * @export
 * @interface AddressTransactionEventListResponse
 */
export interface AddressTransactionEventListResponse {
    /**
     * 
     * @type {number}
     * @memberof AddressTransactionEventListResponse
     */
    limit: number;
    /**
     * 
     * @type {number}
     * @memberof AddressTransactionEventListResponse
     */
    offset: number;
    /**
     * 
     * @type {number}
     * @memberof AddressTransactionEventListResponse
     */
    total: number;
    /**
     * 
     * @type {Array<object>}
     * @memberof AddressTransactionEventListResponse
     */
    results: Array<object>;
}

export function AddressTransactionEventListResponseFromJSON(json: any): AddressTransactionEventListResponse {
    return AddressTransactionEventListResponseFromJSONTyped(json, false);
}

export function AddressTransactionEventListResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): AddressTransactionEventListResponse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'limit': json['limit'],
        'offset': json['offset'],
        'total': json['total'],
        'results': json['results'],
    };
}

export function AddressTransactionEventListResponseToJSON(value?: AddressTransactionEventListResponse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'limit': value.limit,
        'offset': value.offset,
        'total': value.total,
        'results': value.results,
    };
}
