// @stacks/blockchain-api-client/src/generated/models/BnsGetNamespacePriceResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * Fetch price for namespace.
 * @export
 * @interface BnsGetNamespacePriceResponse
 */
export interface BnsGetNamespacePriceResponse {
    /**
     * 
     * @type {string}
     * @memberof BnsGetNamespacePriceResponse
     */
    units: string;
    /**
     * 
     * @type {string}
     * @memberof BnsGetNamespacePriceResponse
     */
    amount: string;
}

export function BnsGetNamespacePriceResponseFromJSON(json: any): BnsGetNamespacePriceResponse {
    return BnsGetNamespacePriceResponseFromJSONTyped(json, false);
}

export function BnsGetNamespacePriceResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): BnsGetNamespacePriceResponse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'units': json['units'],
        'amount': json['amount'],
    };
}

export function BnsGetNamespacePriceResponseToJSON(value?: BnsGetNamespacePriceResponse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'units': value.units,
        'amount': value.amount,
    };
}
