// @stacks/blockchain-api-client/src/generated/models/NetworkBlockTimeResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * GET request that target block time for a given network
 * @export
 * @interface NetworkBlockTimeResponse
 */
export interface NetworkBlockTimeResponse {
    /**
     * 
     * @type {number}
     * @memberof NetworkBlockTimeResponse
     */
    target_block_time: number;
}

export function NetworkBlockTimeResponseFromJSON(json: any): NetworkBlockTimeResponse {
    return NetworkBlockTimeResponseFromJSONTyped(json, false);
}

export function NetworkBlockTimeResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): NetworkBlockTimeResponse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'target_block_time': json['target_block_time'],
    };
}

export function NetworkBlockTimeResponseToJSON(value?: NetworkBlockTimeResponse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'target_block_time': value.target_block_time,
    };
}
