// @stacks/blockchain-api-client/src/generated/models/TransactionEventsResponse.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Stacks Blockchain API
 * Welcome to the API reference overview for the [Stacks Blockchain API](https://docs.hiro.so/stacks-blockchain-api).  [Download Postman collection](https://hirosystems.github.io/stacks-blockchain-api/collection.json) 
 *
 * The version of the OpenAPI document: STACKS_API_VERSION
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
/**
 * GET event for the given transaction
 * @export
 * @interface TransactionEventsResponse
 */
export interface TransactionEventsResponse {
    /**
     * 
     * @type {number}
     * @memberof TransactionEventsResponse
     */
    limit: number;
    /**
     * 
     * @type {number}
     * @memberof TransactionEventsResponse
     */
    offset: number;
    /**
     * 
     * @type {Array<object>}
     * @memberof TransactionEventsResponse
     */
    results: Array<object>;
}

export function TransactionEventsResponseFromJSON(json: any): TransactionEventsResponse {
    return TransactionEventsResponseFromJSONTyped(json, false);
}

export function TransactionEventsResponseFromJSONTyped(json: any, ignoreDiscriminator: boolean): TransactionEventsResponse {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'limit': json['limit'],
        'offset': json['offset'],
        'results': json['results'],
    };
}

export function TransactionEventsResponseToJSON(value?: TransactionEventsResponse | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'limit': value.limit,
        'offset': value.offset,
        'results': value.results,
    };
}
